src/controller/tests/TestReadChunking.cpp

```diff
@@ -509,13 +622,260 @@ void TestCommandInteraction::TestDynamicEndpoint(nlTestSuite * apSuite, void * a
+    app::InteractionModelEngine::GetInstance()->GetReportingEngine().SetMaxAttributesPerChunk(2);
+
+    DataVersion dataVersionStorage1[ArraySize(testEndpointClusters)];
+    DataVersion dataVersionStorage5[ArraySize(testEndpoint5Clusters)];
+
+    gMutableAttrAccess.Reset();
+
+    // Register our fake dynamic endpoint.
+    emberAfSetDynamicEndpoint(0, kTestEndpointId, &testEndpoint, 0, 0, Span<DataVersion>(dataVersionStorage1));
+    emberAfSetDynamicEndpoint(1, kTestEndpointId5, &testEndpoint5, 0, 0, Span<DataVersion>(dataVersionStorage5));
+
+    {
+        app::AttributePathParams attributePath;
+        app::ReadPrepareParams readParams(sessionHandle);
+
+        readParams.mpAttributePathParamsList    = &attributePath;
+        readParams.mAttributePathParamsListSize = 1;
+        readParams.mMinIntervalFloorSeconds     = 0;
+        readParams.mMaxIntervalCeilingSeconds   = 2;
+
+        //// TEST 1 -- Read using wildcard paths
+        {
+            TestMutableReadCallback readCallback;
+
+            gIterationCount = 1;
+
+            app::ReadClient readClient(engine, &ctx.GetExchangeManager(), readCallback.mBufferedCallback,
+                                       app::ReadClient::InteractionType::Subscribe);
+
+            NL_TEST_ASSERT(apSuite, readClient.SendRequest(readParams) == CHIP_NO_ERROR);
+
+            // When the report engine starts to report attributes in endpoint 5, mark cluster 1 as dirty.
+            // The report engine should NOT include it in initial report to reduce traffic.
+            // We are expected to miss attributes on kTestEndpointId during initial reports.
+            ChipLogProgress(DataManagement, "Case 1-1: Set dirty during priming report.");
+            readCallback.mActionOn[AttrOnEp5<Attr1>] = TouchAttrOnEndpoint1<Attr1>;
+            DriveIOUntilSubscriptionEstablished(&readCallback);
+            CheckValues(&readCallback, { { AttrOnEp1<Attr1>, 1 } });
+
+            ChipLogProgress(DataManagement, "Case 1-1: Check for attributes missed last report.");
+            DoTest(&readCallback, { .chunksize = 2, .expectedValues = { { AttrOnEp1<Attr1>, 2 } } });
+
+            ChipLogProgress(DataManagement, "Case 1-2: Set dirty during chunked report by wildcard path.");
+            readCallback.mActionOn[AttrOnEp5<Attr2>] = WriteAttrOnEndpoint5<Attr3, 3>;
+            DoTest(&readCallback,
+                   { .chunksize = 2,
+                     .preworks = { WriteAttrOnEndpoint5<Attr1, 2>, WriteAttrOnEndpoint5<Attr2, 2>, WriteAttrOnEndpoint5<Attr3, 2> },
+                     .expectedValues                = { { AttrOnEp5<Attr1>, 2 }, { AttrOnEp5<Attr2>, 2 }, { AttrOnEp5<Attr3>, 3 } },
+                     .attributesWithSameDataVersion = { { AttrOnEp5<Attr1>, AttrOnEp5<Attr2>, AttrOnEp5<Attr3> } } });
+
+            ChipLogProgress(DataManagement, "Case 1-3: Set dirty during chunked report by wildcard path -- new dirty attribute.");
```

Comment:
I'm not sure what the case numbering scheme here is, since the top number "1" is never changed at all.

I'd rather sub-steps as part of the same logical test be sub-numbered, so the above cases become "1-1, 1-2, 2-1, 3-1, 4-1".